packages/fetchai/skills/oracle_aggregation/handlers.py

```diff
@@ -0,0 +1,283 @@
+            OefSearchDialogues, self.context.oef_search_dialogues
+        )
+        oef_search_dialogue = cast(
+            Optional[OefSearchDialogue], oef_search_dialogues.update(oef_search_msg)
+        )
+        if oef_search_dialogue is None:
+            self._handle_unidentified_dialogue(oef_search_msg)
+            return
+
+        # handle message
+        if oef_search_msg.performative is OefSearchMessage.Performative.OEF_ERROR:
+            self._handle_error(oef_search_msg, oef_search_dialogue)
+        elif oef_search_msg.performative is OefSearchMessage.Performative.SEARCH_RESULT:
+            self._handle_search(oef_search_msg, oef_search_dialogue)
+        else:
+            self._handle_invalid(oef_search_msg, oef_search_dialogue)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_unidentified_dialogue(self, oef_search_msg: OefSearchMessage) -> None:
+        """
+        Handle an unidentified dialogue.
+
+        :param msg: the message
+        """
+        self.context.logger.info(
+            "received invalid oef_search message={}, unidentified dialogue.".format(
+                oef_search_msg
+            )
+        )
+
+    def _handle_error(
+        self, oef_search_msg: OefSearchMessage, oef_search_dialogue: OefSearchDialogue
+    ) -> None:
+        """
+        Handle an oef search message.
+
+        :param oef_search_msg: the oef search message
+        :param oef_search_dialogue: the dialogue
+        :return: None
+        """
+        self.context.logger.info(
+            "received oef_search error message={} in dialogue={}.".format(
+                oef_search_msg, oef_search_dialogue
+            )
+        )
+
+    def _handle_search(
+        self, oef_search_msg: OefSearchMessage, oef_search_dialogue: OefSearchDialogue
+    ) -> None:
+        """
+        Handle the search response.
+
+        :param agents: the agents returned by the search
+        :return: None
+        """
+        if len(oef_search_msg.agents) == 0:
+            self.context.logger.info(
+                f"found no agents in dialogue={oef_search_dialogue}, continue searching."
+            )
+            return
+        strategy = cast(GenericStrategy, self.context.strategy)
+        self.context.logger.info(
+            "found agents={}.".format(
+                list(map(lambda x: x[-5:], oef_search_msg.agents)),
+            )
+        )
+        consensus_dialogues = cast(ConsensusDialogues, self.context.consensus_dialogues)
+        strategy.add_peers(oef_search_msg.agents)
+        for counterparty in strategy.peers:
+            obs_msg, _ = consensus_dialogues.create(
+                counterparty=counterparty,
+                performative=ConsensusMessage.Performative.OBSERVATION,
+                value=1,
+                time=1,
+                source="source",
+                signature="sign",
```

Comment:
I'd define a helper method somewhere which returns this. Then later you just need to update the method implementation.